### PR TITLE
fix mainReg in 21 regions cs2

### PR DESCRIPTION
**Pull request by chroetz**
# Purpose of this PR

* the compareScenarios2 scripts are refactored.
* profiles.csv is replaced by profiles.json, which is much more readable and usable
* a bug in the cs2-profile *AriadneDEU* is fixed
* some cs2-profiles are renamed (*default* -> *H12*), (*short* -> *H12-short*)
* new cs2-profiles for 21 region runs are added (*EU27*, *EU27-short*)
* an example for a cs2-profile with a user section is added (*mySection*)
* the default choice of cs2-profiles is improved
* the default choice of cs2-profiles is highlighted
* new function `isSlurmAvailable()` in new script `./scripts/utils/isSlurmAvailable.R` is added

## Type of change

- [X] Bug fix 
- [X] Refactoring

## Checklist:

- [X] My code follows the coding etiquette
- [X] I have performed a self-review of my own code
- [X] Changes are commented, particularly in hard-to-understand areas
- [X] I have updated the in-code documentation
- [X] I have adjusted reporting where it was needed
- [X] The model compiles and runs successfully (`Rscript start.R -q`)

## Further information:

Test runs are here: `/p/tmp/cschoetz/cs2fix/remind`
For all available cs2-profiles, compareScenarios2 was executed on 4 REMIND runs (2x 21 regions, 2x 12 regions). The results can be found in that folder.

### CODE REVIEW

**Review by robinhasse — APPROVED** [review]
With this PR, the script finished successfully where it crashed before. Thanks Christof!

File: scripts/cs2/run_compareScenarios2.R [hunk 1]
```diff
@@ -56,9 +50,7 @@ run_compareScenarios2 <- function(
     cfgDefault = defaultConfigPath,
     outputDir = outfilepath,
     outputFile = outFileName,
-    outputFormat = "PDF",
-    reg = regionList,
-    mainReg = mainRegName
+    outputFormat = "PDF"
```
Comment (chroetz) [inline scripts/cs2/run_compareScenarios2.R:53]: `outputFormat = "PDF"` is the default of `remind2::compareScenarios2()` and does not need to be set here

File: scripts/cs2/run_compareScenarios2.R [hunk 2]
```diff
@@ -68,21 +60,22 @@ run_compareScenarios2 <- function(
...
   ) {
-    message("applying profile ", profileName)
-    profile <- as.list(profiles[profiles$name == profileName, ])
-    profile$name <- NULL
-    profile <- lapply(profile, trimws)
-    profile <- profile[vapply(profile, function(s) nchar(s) > 0, logical(1))]
-    profileEval <- lapply(
+    message("Applying profile ", profileName)
+    profile <- profiles[[profileName]]
+    profileEval <- lapply( # evaluate entries of profile as R code
```
Comment (chroetz) [inline scripts/cs2/run_compareScenarios2.R:65]: add comment about `.`

File: scripts/cs2/run_compareScenarios2.R [hunk 3]
```diff
@@ -68,21 +60,22 @@ run_compareScenarios2 <- function(
...
     )
     args[names(profile)] <- profileEval
   } else {
-    message("using default profile")
+    message("Using default profile")
   }
+  
+  message("Use following cs2 settings:")
+  for (i in seq_along(args)) 
+    message("  ", names(args)[i], " = ", capture.output( dput(args[[i]])))
```
Comment (chroetz) [inline scripts/cs2/run_compareScenarios2.R:78]: unnecessary space. check linter

File: scripts/output/comparison/compareScenarios2.R [hunk 4]
```diff
@@ -4,7 +4,20 @@
...
-# ---- Define set of runs that will be compared ----
+
+
+
+
+# Header ------------------------------------------------------------------
+
+
+# Make sure the script is called via output.R.
+if (!exists("outputdirs")) stop("outputdirs does not extist. Please call comapreScenarios.R via output.R, which defines outputdirs.")
```
Comment (chroetz) [inline scripts/output/comparison/compareScenarios2.R:15]: define what variables are expected to exist here. Maybe allow to load them from command line?

File: scripts/output/comparison/compareScenarios2.R [hunk 5]
```diff
@@ -165,48 +115,32 @@ startComp <- function(
...
-    }
 
-  }
+# Code --------------------------------------------------------------------
+
+
+# load cs2 profiles
+profilesFilePath <- normalizePath("./scripts/cs2/profiles.json")
+profiles <- jsonlite::read_json(profilesFilePath, simplifyVector = FALSE)
+profiles <- profiles[!startsWith(names(profiles), "_")] # remove comments
```
Comment (chroetz) [inline scripts/output/comparison/compareScenarios2.R:126]: p[rofiles <- readProfiles should be a function of some package

File: scripts/output/comparison/compareScenarios2.R [hunk 6]
```diff
@@ -39,124 +59,54 @@ chooseFromSequence <- function(sequence, title, default) {
...
+# Find a suitable default cs2 profile depending on config.RData.
+determineDefaultProfiles <- function(outputDir) {
+  env <- new.env()
+  load(file.path(outputDir, "config.Rdata"), envir = env)
+  regionMappingFile <- basename(env$cfg$regionmapping)
+  defaults <- switch(
+    regionMappingFile,
+    "default",
+    "regionmappingH12.csv" =  c("H12", "H12-short"),
+    "regionmapping_21_EU11.csv" =  c("H12", "H12-short", "EU27", "EU27-short", "AriadneDEU"))
```
Comment (chroetz) [inline scripts/output/comparison/compareScenarios2.R:72]: hard coded parameters...

File: scripts/cs2/run_compareScenarios2.R [hunk 7]
```diff
@@ -8,27 +8,21 @@ library(lucode2) # getScenNames
...
-    text = readLines(profilesFilePath, warn = FALSE),
-    header = TRUE,
-    sep = ";",
-    colClasses = "character",
-    comment.char = "#",
-    quote = "")
+  # load cs2 profiles
+  profilesFilePath <- normalizePath("./scripts/cs2/profiles.json")
+  profiles <- jsonlite::read_json(profilesFilePath, simplifyVector = FALSE)
+  profiles <- profiles[!startsWith(names(profiles), "_")] # remove comments
```
Comment (chroetz) [inline scripts/cs2/run_compareScenarios2.R:25]: should be a function `readProfiles()`

File: scripts/cs2/run_compareScenarios2.R (hunk 3 above)
Comment (chroetz) [reply scripts/cs2/run_compareScenarios2.R:78]: Just message full call to `remind2::compareScenarios2()`?

**Review by 0UmfHxcvx5J7JoaOhFSs5mncnisTJJ6q — APPROVED** [review]

**Review by LaviniaBaumstark — APPROVED** [review]

**Pull request merged**